tasks/human_eval_087.rs

```diff
@@ -5,11 +5,147 @@ HumanEval/87
 /*
 ### VERUS BEGIN
 */
+use std::vec::Vec;
 use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+/// coords is sorted by row in ascending order.
+pub open spec fn row_sorted_asc(coords: Seq<(usize, usize)>) -> bool {
+    forall|i: usize, j: usize|
+        0 <= i < j < coords.len() ==> #[trigger] coords[i as int].0 <= #[trigger] coords[j as int].0
+}
+
+/// In coords, the coordinates of same row are sorted by column in descending order.
+pub open spec fn row_col_sorted_desc(coords: Seq<(usize, usize)>) -> bool {
+    forall|i: usize, j: usize|
+        0 <= i < j < coords.len() && #[trigger] coords[i as int].0 == #[trigger] coords[j as int].0
+            ==> coords[i as int].1 > coords[j as int].1
+}
+
+/// coords contains all coordinates matching x in lst within the given row and at or after the given column.
+pub open spec fn coords_complete_for_row_until_col(
+    lst: Seq<Seq<i32>>,
+    x: i32,
+    coords: Seq<(usize, usize)>,
+    row: usize,
+    col: usize,
+) -> bool
+    recommends
+        0 <= row < lst.len(),
+{
+    forall|j: usize|
+        #![trigger lst[row as int][j as int]]
+        0 <= j < lst[row as int].len() && col <= j && lst[row as int][j as int] == x ==> exists|
+            k: int,
+        |
+            #![trigger coords[k]]
+            0 <= k < coords.len() && coords[k] == (row, j)
+}
+
+/// coords contains all coordinates matching x in lst before the given row.
+pub open spec fn coords_complete_until_row(
+    lst: Seq<Seq<i32>>,
+    x: i32,
+    coords: Seq<(usize, usize)>,
+    row: usize,
+) -> bool {
+    forall|i: usize|
+        #![trigger lst[i as int]]
+        0 <= i < lst.len() && i < row ==> coords_complete_for_row_until_col(lst, x, coords, i, 0)
+}
+
+/// coords contains all coordinates matching x in lst.
+pub open spec fn coords_complete(lst: Seq<Seq<i32>>, x: i32, coords: Seq<(usize, usize)>) -> bool {
+    coords_complete_until_row(lst, x, coords, lst.len() as usize)
+}
+
+/// All points in coords represent occurrences of x in lst.
+pub open spec fn coords_sound(lst: Seq<Seq<i32>>, x: i32, coords: Seq<(usize, usize)>) -> bool {
+    forall|i: usize, j: usize|
+        #![trigger lst[i as int][j as int]]
+        #![trigger coords.contains((i,j))]
+        0 <= i < lst.len() && 0 <= j < lst[i as int].len() && coords.contains((i, j))
+            ==> lst[i as int][j as int] == x
+}
+
+/// coords exactly matches all of the occurrences of x in lst.
+pub open spec fn coords_matches_lst(
+    lst: Seq<Seq<i32>>,
+    x: i32,
+    coords: Seq<(usize, usize)>,
+) -> bool {
+    &&& coords_complete(lst, x, coords)
+    &&& coords_sound(lst, x, coords)
+}
+
+/// Returns lst as a Seq.
+pub open spec fn lst_seq_refl(lst: Vec<Vec<i32>>) -> Seq<Seq<i32>> {
```

Comment:
This isn't very discoverable at the moment, unfortunately, but I think this function is equivalent to doing `lst.deep_view()`.  Looking at the source for the [DeepView](https://verus-lang.github.io/verus/verusdoc/vstd/view/trait.DeepView.html) trait shows more details.

Reply:
If changing to `deep_view` breaks a bunch of proofs, it's probably not worth changing at this point.  Just a note for the future.